### ISSUE TITLE
G1 model lies flat on the ground after GVHMR-to-MinkRetarget conversion

### ISSUE THREAD

**Issue by boweili666** [TeleHuman/PBHC#9]
Hi, I'm using the GVHMR demo to generate motion data. I successfully converted the output .pt into .npz format (compatible with MinkRetarget).
However, when I visualize the result using the MinkRetarget visualization pipeline with the G1 robot, the character is lying flat on the ground, as if Z-up / Y-up mismatch or incorrect upright alignment.

![Image](https://github.com/user-attachments/assets/e2b98f88-f3c2-496d-9216-28fddb288f0f)

**Comment by boweili666** [comment]
here is my convert code

import torch
import numpy as np
import os

# ---------- Paths ----------
pt_file = '/home/lbw/g1/GVHMR/outputs/demo/tennis/hmr4d_results.pt'
output_npz = '/home/lbw/g1/test/video_motion/motion.npz'   # Save to the directory your pipeline expects
# ---------------------------

# Load the .pt file
data = torch.load(pt_file, map_location='cpu')

# Extract SMPL parameters (camera coordinate frame)
smpl_params_global = data['smpl_params_incam']

# Convert every tensor to a NumPy array
np_data = {k: v.cpu().numpy() if torch.is_tensor(v) else v
           for k, v in smpl_params_global.items()}

# --- Rebuild the pose array (global_orient + body_pose + 6 zeros) ---
if 'global_orient' in np_data and 'body_pose' in np_data:
    N = np_data['body_pose'].shape[0]
    global_orient = np_data['global_orient'].reshape(N, 3)   # [N,3]
    body_pose     = np_data['body_pose'].reshape(N, 63)      # [N,63]
    zeros6        = np.zeros((N, 6), dtype=body_pose.dtype)  # [N,6] padding
    poses         = np.concatenate([global_orient, body_pose, zeros6], axis=1)
    print(f"[INFO] poses constructed: {poses.shape}")
else:
    raise ValueError("Missing 'global_orient' or 'body_pose' in the input!")

# ✅ Rename field 'transl' → 'trans' (if it exists)
if 'transl' in np_data:
    np_data['trans'] = np_data.pop('transl')

# Add new fields required downstream
np_data['poses']           = poses      # (N, 72) full SMPL pose in axis-angle
np_data['mocap_framerate'] = 30         # FPS
np_data['gender']          = 'neutral'  # or 'male' / 'female' if known

# Remove the original fields you no longer need
np_data.pop('body_pose',    None)
np_data.pop('global_orient', None)

# Save everything to .npz
np.savez(output_npz, **np_data)
print(f"[SAVE] Converted .pt → .npz with field 'transl' renamed to 'trans': {output_npz}")

**Comment by boweili666** [comment]
here is my convert code
`import torch
import numpy as np
import os

# ---------- Paths ----------
pt_file = '/home/lbw/g1/GVHMR/outputs/demo/tennis/hmr4d_results.pt'
output_npz = '/home/lbw/g1/test/video_motion/motion.npz'   # Save to the directory your pipeline expects
# ---------------------------

# Load the .pt file
data = torch.load(pt_file, map_location='cpu')

# Extract SMPL parameters (camera coordinate frame)
smpl_params_global = data['smpl_params_incam']

# Convert every tensor to a NumPy array
np_data = {k: v.cpu().numpy() if torch.is_tensor(v) else v
           for k, v in smpl_params_global.items()}

# --- Rebuild the pose array (global_orient + body_pose + 6 zeros) ---
if 'global_orient' in np_data and 'body_pose' in np_data:
    N = np_data['body_pose'].shape[0]
    global_orient = np_data['global_orient'].reshape(N, 3)   # [N,3]
    body_pose     = np_data['body_pose'].reshape(N, 63)      # [N,63]
    zeros6        = np.zeros((N, 6), dtype=body_pose.dtype)  # [N,6] padding
    poses         = np.concatenate([global_orient, body_pose, zeros6], axis=1)
    print(f"[INFO] poses constructed: {poses.shape}")
else:
    raise ValueError("Missing 'global_orient' or 'body_pose' in the input!")

# ✅ Rename field 'transl' → 'trans' (if it exists)
if 'transl' in np_data:
    np_data['trans'] = np_data.pop('transl')

# Add new fields required downstream
np_data['poses']           = poses      # (N, 72) full SMPL pose in axis-angle
np_data['mocap_framerate'] = 30         # FPS
np_data['gender']          = 'neutral'  # or 'male' / 'female' if known

# Remove the original fields you no longer need
np_data.pop('body_pose',    None)
np_data.pop('global_orient', None)

# Save everything to .npz
np.savez(output_npz, **np_data)
print(f"[SAVE] Converted .pt → .npz with field 'transl' renamed to 'trans': {output_npz}")
`

**Comment by boweili666** [comment]
Here is my convert code

import torch
import numpy as np
import os

pt_file = '/home/lbw/g1/GVHMR/outputs/demo/tennis/hmr4d_results.pt'
output_npz = '/home/lbw/g1/test/video_motion/motion.npz'   # Save to the directory your pipeline expects

data = torch.load(pt_file, map_location='cpu')

smpl_params_global = data['smpl_params_incam']

np_data = {k: v.cpu().numpy() if torch.is_tensor(v) else v
           for k, v in smpl_params_global.items()}

if 'global_orient' in np_data and 'body_pose' in np_data:
    N = np_data['body_pose'].shape[0]
    global_orient = np_data['global_orient'].reshape(N, 3)   # [N,3]
    body_pose     = np_data['body_pose'].reshape(N, 63)      # [N,63]
    zeros6        = np.zeros((N, 6), dtype=body_pose.dtype)  # [N,6] padding
    poses         = np.concatenate([global_orient, body_pose, zeros6], axis=1)
    print(f"[INFO] poses constructed: {poses.shape}")
else:
    raise ValueError("Missing 'global_orient' or 'body_pose' in the input!")

if 'transl' in np_data:
    np_data['trans'] = np_data.pop('transl')

np_data['poses']           = poses      # (N, 72) full SMPL pose in axis-angle
np_data['mocap_framerate'] = 30         # FPS
np_data['gender']          = 'neutral'  # or 'male' / 'female' if known

np_data.pop('body_pose',    None)
np_data.pop('global_orient', None)

np.savez(output_npz, **np_data)
print(f"[SAVE] Converted .pt → .npz with field 'transl' renamed to 'trans': {output_npz}")

**Comment by boweili666** [comment]
Correct body motion but incorrect root orientation after GVHMR-to-MinkRetarget conversion

**Comment by boweili666** [comment]
my phc_retarget get similar results

**Comment by Songzhanjie-31608** [comment]
另外还想问一下,为什么要在poses后面每帧加6个0,我看作者说SMPL的标准格式里poses不是(frames,66)吗

**Comment by openhe-hub** [comment]
> yes,I modify your code ,changed the shape of betas and poses ,and then problem has been solved Thanks for your reply Best wishes!

Hi I have the same problem. Can you share your code?

**Comment by Songzhanjie-31608** [comment]
> Correct body motion but incorrect root orientation after GVHMR-to-MinkRetarget conversion

Hi,Did you try the motion_filter  in the 'smpl_retarget'?
the data obtained from GVHMR maybe have some errors,but I met many problems when trying the motion_filter.
If you have got it,could you help me a little?
Best wishes!

**Comment by git33jy** [comment]
您好，请问这个用于重定向使用的pkl文件是如何生成的
